pkg/hooks/manager.go

```diff
@@ -161,15 +161,17 @@ func processSideCarSocket(socketPath string) (*callBackClient, bool, error) {
 		versionsSet[version] = true
 	}
 
-	// The order matters. We should match newer versions first.
-	supportedVersions := []string{
-		hooksV1alpha3.Version,
-		hooksV1alpha2.Version,
-		hooksV1alpha1.Version,
+	supportedVersions := map[string]bool{
+		hooksV1alpha3.Version: hooksV1alpha3.Deprecated,
```

Comment:
Doesn't this mark v1alpha3 as deprecated too?

Reply:
@0xFelix Deprecated is a boolean, for v1alpha3 it is `false`, the other two is `true`

Reply:
Ah! My mistake, this pattern caught me off guard. What about a list of deprecated versions that you only maintain in `manager.go`? You need to touch this file anyways when adding a new API version. This way you would get a direct overview of deprecated versions when looking at the code?

Reply:
Sorry, I missed your comment.

> What about a list of deprecated versions that you only maintain in manager.go?

I can add it, I think it would be beneficial to avoid doing:

```Go
               // Check for deprecated APIs being used.
               hookVersions := map[string]bool{
                       hooksv1alpha3.Version: hooksv1alpha3.Deprecated,
                       hooksv1alpha2.Version: hooksv1alpha2.Deprecated,
                       hooksv1alpha1.Version: hooksv1alpha1.Deprecated,
               }
```

but I think the deprecation value should still be part of each version module, like the Version string.

Reply:
https://github.com/victortoso/kubevirt/blob/deprecate-hooks-v1alphav1/pkg/hooks/manager.go#L59